contracts/PoolManager.sol

```diff
@@ -500,8 +500,10 @@ contract PoolManager is IPoolManager, Owned, NoDelegateCall, ERC1155, IERC1155Re
 
         /// @solidity memory-safe-assembly
         assembly {
+            let valuePtr := add(value, 32)
             for { let i := 0 } lt(i, nSlots) { i := add(i, 1) } {
-                mstore(add(value, mul(add(i, 1), 32)), sload(add(startSlot, i)))
+                mstore(valuePtr, sload(add(startSlot, i)))
+                valuePtr := add(valuePtr, 32)
             }
```

Comment:
Consider this
```Solidity
            let valuePtr := add(value, 32)
            let endSlot := add(startSlot, nSlots)
            for {} 1 {} {
                mstore(valuePtr, sload(startSlot))
                valuePtr := add(valuePtr, 32)
                startSlot := add(startSlot, 1)
                if eq(startSlot, endSlot) { break }
            }
```
after which `poolExtsloadTickInfoStruct.snap` becomes 2704. It saves one `add` per loop iteration and seemingly involves less stack shuffling.

Furthermore, after replacing
```Solidity
bytes memory value = new bytes(32 * nSlots);
```
with
```Solidity
bytes memory value = new bytes(nSlots << 5);
```
the gas comes down to 2625.

Reply:
Thanks, I wasn't 100% convinced I couldn't get rid of `i` but didn't figure it out! The first recommendation makes sense, but I imagined the overflow protection is desired for `bytes memory value = new bytes(32 * nSlots)`. I'll see what someone from Uniswap says

Reply:
Someone has to pass `nSlots >= 1 << 251` for it to overflow though, in which case gas would run out in the loop.